pages/blog/graphql-error-handling-with-fp.mdx

```diff
@@ -0,0 +1,631 @@
+import withArticle from '../../ui/blog/article'
+
+export const meta = {
+  title: 'GraphQL error handling to the max with Typescript, codegen and fp-ts',
+  author: 'Ghislain Thau',
+  tags: ['graphql', 'typescript', 'codegen', 'fp-ts'],
+  date: '2022-02-17',
+  description:
+    'Handle unsafe APIs in a typesafe way with Typescript and Functional Programming',
+}
+
+export default withArticle({ ...meta })
+
+Contrary to REST, GraphQL does not use Status Code to differentiate a successful result from an exception. In REST, one can send a response with a appropriate status code to inform the caller about the kind of exception encountered, e.g.:
+- 400: bad request (e.g. invalid inputs)
+- 401 or 403: unauthorized (need authentication) or forbidden (insufficient permission)
+- 404: resource not found
+- 500: internal server error
+- and more.
+
+
+# GraphQL default error handling
+
+
+In GraphQL, you'll get a 5xx error in case the server is not available at all, but for other exceptions you'll get the standard error handling mechanism from GraphQL: in the JSON you receive there is:
```

Comment:
```suggestion
In GraphQL, all requests must go to a single endpoint:


> POST https://example.com/graphql

A GraphQL API will always return a `200 OK` Status Code, even in case of error.
You'll get a 5xx error in case the server is not available at all.

The standard error handling mechanism from GraphQL with return a JSON containing:
```